.pre-commit-config.yaml

```
@@ -54,6 +54,20 @@ repos:
         types: [python]
         files: "^backend/capellacollab"
         exclude: "^backend/capellacollab/alembic/"
+      - id: mypy
+        name: mypy
+        entry: mypy
+        args: [--config-file=./backend/pyproject.toml]
+        language: python
+        types: [python]
+        additional_dependencies:
+          - mypy
+          - fastapi
+          - pydantic
+          - sqlalchemy
+        files: "^backend/capellacollab"
+        exclude: "^backend/capellacollab/alembic/"
+        verbose: true
```

Why not use the hook provided by pre-commit?

```suggestion
  - repo: https://github.com/pre-commit/mirrors-mypy
    rev: v1.5.1
    hooks:
      - id: mypy
        types_or: [python, spec]
        files: "^backend/capellacollab"
        exclude: "^backend/capellacollab/alembic/"
        args: [--config-file=./backend/pyproject.toml]
        additional_dependencies:
         - fastapi
         - pydantic
         - sqlalchemy
```

I expect then you can drop the additional dependencies.

---

I tried this as the very first thing, but switched to the local version as it didn't work properly. However, it seems it was just missing the extra dependencies to work properly (quite annoying to have them, but apparently they can't take the dependencies from the toml).
So switched to the mirror but kept the `additional_dependencies`.